platform-hub-api/spec/mailers/announcement_mailer_spec.rb

```diff
@@ -3,48 +3,89 @@
 RSpec.describe AnnouncementMailer, type: :mailer do
 
   describe '.announcement_email' do
-    context 'a very simple one line announcement' do
-      let(:from) { Rails.application.config.email_from_address }
-
-      let :recipients do
-        [
-          'foo@example.org',
-          'bar@example.org'
-        ]
-      end
+    let(:from) { Rails.application.config.email_from_address }
 
-      let(:text) { 'foo bar'}
-      let(:announcement) { create :announcement, text: text }
+    let :recipients do
+      [
+        'foo@example.org',
+        'bar@example.org'
+      ]
+    end
 
-      let :expected_subject do
-        "[#{announcement.level}] #{announcement.title}"
-      end
-      let :expected_text do
-        "#{text}\n\n"  # Because we have extra newlines in the template
-      end
-      let :expected_html do
-        "<div>#{text}</div>\n<br><br>"
-      end
+    let :text do
+      <<~TEXT
+        Hello,
 
-      before do
-        @email = AnnouncementMailer.announcement_email announcement, recipients
-      end
+        This is an email announcement of epic foo proportions…
 
-      it 'queues the email' do
-        assert_emails 1 do
-          @email.deliver_now
-        end
-      end
+        - foo
+        - bar
+        - baz
 
-      it 'renders the email (both in text and html)' do
-        expect(@email[:from].value).to eq from
-        expect(@email[:to].value).to eq from
-        expect(@email.bcc).to eq recipients
-        expect(@email.subject).to eq expected_subject
-        expect(@email.text_part.body.to_s).to eq expected_text
-        expect(@email.html_part.body.to_s).to match expected_html
+        Don't miss out: https://foo.com
+
+        Faithfully yours,
+        The Foo Team
+      TEXT
+    end
+
+    let(:announcement) { create :announcement, text: text }
+
+    let :expected_subject do
+      "[#{announcement.level}] #{announcement.title}"
+    end
+
+    let :expected_text do
+      "#{text}\n\n"  # Because we have extra newlines in the template
+    end
+
+    let :expected_html do
+      <<~HTML
+        <!DOCTYPE html>
+        <html>
+          <head>
+            <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+          </head>
+
+          <body>
+            <p>Hello,</p>
+
+        <p>This is an email announcement of epic foo proportions…</p>
+
+        <p>- foo
+        <br />- bar
+        <br />- baz</p>
+
+        <p>Don't miss out: <a href="https://foo.com">https://foo.com</a></p>
+
+        <p>Faithfully yours,
+        <br />The Foo Team
+        </p>
+        <br><br>
+
+          </body>
+        </html>
+      HTML
+    end
+
+    before do
+      @email = AnnouncementMailer.announcement_email announcement, recipients
+    end
+
+    it 'queues the email' do
+      assert_emails 1 do
+        @email.deliver_now
       end
     end
+
+    it 'renders the email (both in text and html)' do
+      expect(@email[:from].value).to eq from
```

Comment:
Unfortunately not, due to https://github.com/rails/rails/issues/7602#issuecomment-263771279